docs/1-introduction/1-1-getting-started/1-1-3-hello-triangle.md

```diff
@@ -1,11 +1,456 @@
+
+We use this stage to apply most of our shading techniques, from basic lighting, to textures and shadows, all the way to physically based rendering.
+
+Since we did not specify any shader between the VS and the PS, our input here is the output of the VS, and the output is one or more render targets.
+
+Let's look at our Pixel Shader now:
+#### Main.ps.hlsl
+```hlsl
+struct PSInput
+{
+    float4 position: SV_POSITION;
+    float3 color: COLOR0;
+};
+
+struct PSOutput
+{
+    float4 color: SV_TARGET0;
+};
+
+PSOutput Main(PSInput input)
+{
+    PSOutput output = (PSOutput)0;
+    output.color = float4(input.color, 1.0);
+    return output;
+}
+```
+Here as well we have an input `PSInput`, and an output `PSOutput`.
+
+Since our input here is the VS's output, `PSInput` must match **exactly** the `VSOutput` in vertex shader, if the layout differ even a little, for example by moving `float3 color` above `float4 position`, the D3D11's debug layer will scream at us, and the shaders will fail to compile.
+
+Next we have our output, D3D11 gives us the possibility to write to multiple render targets, but we are not doing that, so we'll just be writing a `float4` as our output, which is an RGBA color.
+
+Notice how we have another semantic string attached to the `color` field, this semantic string specifies which render target we want to be writing to, the `0` after `SV_TARGET` is the index of our render target, in our case, we have only one so we write `SV_TARGET0` or `SV_TARGET`.
+
+And lastly, our `Main` function, following the same pattern as in the VS, we have one parameter, the input, and one return value, the output, again we create an instance of `PSOutput`, initialize everything to 0, and write the color we got from the input, to our output.
+## Compiling shaders
+
+Now that we wrote our shader code and saved it somewhere, we have to feed this to the GPU, to do that we'll have our D3DCompiler get to work.
+
+First, we will declare some functions that will help us compile our shaders more quickly.
+
+### HelloTriangle.hpp
+```cpp
+bool CompileShader(
+    const std::wstring_view fileName,
+    const std::string_view entryPoint,
+    const std::string_view profile,
+    ComPtr<ID3D10Blob>& shaderBlob) const;
+
+[[nodiscard]] ComPtr<ID3D11VertexShader> CreateVertexShader(
+    const std::wstring_view fileName,
+    ComPtr<ID3D10Blob>& vertexShaderBlob) const;
+
+[[nodiscard]] ComPtr<ID3D11PixelShader> CreatePixelShader(std::wstring_view fileName) const;
+```
+In order we have:
+
+`CompileShader`: This function is the core for compiling shaders, it requires 3 input parameters:
+
+- `fileName`: is the path of the shader file we want to compile.
+- `entryPoint`: is the name of the function where the shader begins execution.
+- `profile`: which is basically the version of HLSL we want to use, the higher the profile number, the more features there are.
```

Comment:
maybe add another sentence here that it describes the so called Shader Model, and for D3D11 we will be sticking with SM5.0 (there should be an msdn link somewhere pointing to what SM means and is and maybe even an overview of what  capabilities each SM offers)